### DISCUSSION TITLE
Use an icon in the nav section

### DISCUSSION THREAD

**Discussion by NikosAlexandris** [squidfunk/mkdocs-material#7326] [Q&A]
Is it possible to use a custom icon in the nav part of the configuration file `mkdocs.yml` ?

Example:

``` yaml
nav:
  - Overview:
    - :material/some-icon: Overview: index.md
```

or

``` yaml
nav:
  - Overview:
    - :some-icon: Overview: index.md
```

?

**Comment by kamilkrzyskow** [comment]
Hello @NikosAlexandris,
this syntax `:icon:` is only processed in the Markdown contents. To add icons to your page titles you can use the `icon` meta key:
- https://squidfunk.github.io/mkdocs-material/reference/#setting-the-page-icon

Here is an example:
- [icons-test-for-NikosAlexandris.zip](https://github.com/user-attachments/files/16082800/icons-test-for-NikosAlexandris.zip)
it uses a custom icon, but you can set the path with `/` to any icons defined in the material theme package:
    - https://github.com/squidfunk/mkdocs-material/tree/master/material/templates/.icons

**Reply by NikosAlexandris** [reply]
Thank you. This does not directly answer the original question though. I have some Notebooks which I'd like to prepend with an icon where their title appears in the sidebar and then another pytest coverage HTML artefact which is also not a Markdown file per se. Hence, it's not possible to use the standard way of defining icon-paths inthe YAML-styled header inside a Markdown file.

Naively, in the `mkdocs.yml` file and in the `nav` section, I was trying something like :

``` yaml
    - 󰦕 Test Coverage: coverage.md
```

which is _followed_ by

``` yaml
  - coverage:
      html_report_dir: docs/development/coverage_html
```

**Reply by alexvoss** [reply]
@kamilkrzyskow had it right. The navigation in `mkdocs.yml` is not processed as Markdown. While including HTML tags might give you the desired effect when combined with some CSS, at the moment the HTML would also appear in places where it does not belong, such as in the title element of the page header. Give it a try and you will see. The definition and handling of titles with more than textual content is not an easy topic, as you can see from this exchange: https://github.com/mkdocs/mkdocs/issues/3357. 

Perhaps this is something that could be achieved with just some CSS if you have some way of matching the navigation elements in question?

**Reply by kamilkrzyskow** [reply]
I don't really understand the coverage part, how are those 2 values connected to each other and nav in general?
Is the `coverage` value a plugin? 
If so, then in what way does the `docs/development/coverage_html` have to do with anything concerning navigation?

> This does not directly answer the original question though

You have asked specifically for adding an icon and provided an example, so I explained why it's not possible.

But answering the question in general. Yes, you can use text based Unicode emojis:
```yaml
nav:
  - Home: index.md
  - Section:
    - 🤖 Test coverage: test/index.md
```

but those change the appearance based on the rendering software like Chrome / Firefox browser, PC / Mobile etc.
So I didn't even consider them as a valid approach.

---

Another approach would be to use customization:
- https://squidfunk.github.io/mkdocs-material/customization/#extending-the-theme

and modify how nav-items are rendered, and add some custom logic there to load the SVG icons based on a criteria like the file name or something else. You can look at the `icon` setting implementation to see how the syntax works:
https://github.com/squidfunk/mkdocs-material/blob/c7c8fcb3af8390e15fc1de4cdf24bd28bf989f19/src/templates/partials/nav-item.html#L44-L47

---

Like Alex mentioned above you could try to use CSS to target navigation items based on their href and maybe use `:after``:before` classes to add the icons, but I'm not sure if this will work 🤔 
- https://github.com/squidfunk/mkdocs-material/discussions/6547#discussioncomment-7917358

---

You could perhaps use some [hook](https://www.mkdocs.org/user-guide/configuration/#hooks) magic for the `coverage` HTML files that are not Markdown files to not include them as-is to the navigation but add a Markdown file with a YAML header and then in the `on_page_content` event add the HTML from those other files. This is likely the worst option, but as I said I don't see the full picture of how everything is set up.
